variables.tf

```
@@ -14,3 +17,26 @@ variable "set_sensitive_values" {
   type        = any
   default     = []
 }
+
+#######
+# IRSA
+#######
+variable "irsa_config" {
+  description = "Input configuration for IRSA module"
+  type = object({
+    role_name                         = optional(string)
+    role_policy_arns                  = optional(map(string), null)
+    create_kubernetes_namespace       = optional(bool, true)
+    create_kubernetes_service_account = optional(bool, true)
+    kubernetes_namespace              = optional(string, "")
+    kubernetes_service_account        = optional(string, "")
+    oidc_providers                    = optional(any, null)
+  })
+  default = {}
+}
+
+variable "tags" {
+  description = "A map of tags to add the the IAM role"
+  type        = map(any)
```

```suggestion
  type        = map(string)
```

AWS tags can only allow string values anyway.